central/notifications/datastore/datastore.go

```diff
@@ -0,0 +1,35 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/central/notifications/datastore/internal/search"
+	"github.com/stackrox/rox/central/notifications/datastore/internal/store"
+	"github.com/stackrox/rox/central/notifications/datastore/internal/writer"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+)
+
+// DataStore provides an interface to handle notifications.
+//
+//go:generate mockgen-wrapper
+type DataStore interface {
+	// AddNotification is intended to be used by the notification handler to add
+	// notifications to the buffered writer. Flush initiates a batched upsert to
+	// the database.
+	AddNotification(ctx context.Context, notification *storage.Notification) error
+	Flush(ctx context.Context) error
+
+	// These provide read-only access and are intended to be used by the API service.
```

Comment:
super nit: personally feel like this comment doesn't really give much value + the datastore _technically_ may be used by any client, not necessarily restricted to the API service.